agent/executor.py

```diff
@@ -122,8 +123,29 @@ def schedule(executor: Executor) -> asyncio.Future:
     :param executor:
     :return: executor.start() wrapped in Future
     """
-    return asyncio.ensure_future(executor.start())
+    awaitable = executor.start()
+    future = asyncio.ensure_future(awaitable)
+    return future
 
 
 def spin():
     asyncio.get_event_loop().run_forever()
+
+
+class Locker:
+    LOCKDIR = '/var/lock'
```

Comment:
We should probably use a lockfile within our directory, because we can't make assumptions on where we can write on certain platforms (such as within a docker container or in a Snap).

Reply:
@vpetersson I was trying to follow https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s09.html .

I'm convinced /var/lock should be available for snaps. I haven't tried it myself. The snap docs say that every snap gets its private `/tmp`, though. But, when was the last time we tried running agent as a snap? Perhaps something else is not allowed, like forking? 

In Docker container `/var/lock` needs to be mounted which I did in the recently merged docker-compose.yml.

Even though our lock file is not shared by other applications it is used by multiple instances of our application. A predefined directory like `/var/lock` already exists when we run our app for the first time. But our directory (like /opt/wott) may or may not exist, depending on how the app was installed. And if it does not exist the app will try to create it, and guess what can happen if two instances try to do that at the same time. So storing the lock in our directory is not a problem, creating the directory is a problem.

If a user installs our app as a deb package, `/opt/wott` will be created at installation. Same for snap. Thus we will only have this non-existing directory issue if our app was installed and run manually on a clean system.

**tl;dr**: I did what you asked for, but not sure if it's right.